### PR TITLE
migrated from jest to vitest

### PR DESCRIPTION
**Pull request by ghalibansari**

### CODE REVIEW

File: vitest.config.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// eslint-disable-next-line import/no-unresolved
```
Comment (ghalibansari) [inline vitest.config.ts:1]: i was facing this issue here @Goldziher

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -1,41 +1,41 @@
-name: "CodeQL"
+name: 'CodeQL'
```
Comment (Goldziher) [inline .github/workflows/codeql.yml:1]: Run `pnpm format`, something in your IDE settings doesn't factor in the prettier config

File: package.json [hunk 3]
```diff
@@ -34,30 +34,30 @@
...
         "eslint": "^8.29.0",
         "husky": "^8.0.2",
-        "jest": "^29.3.1",
         "lint-staged": "^13.1.0",
         "prettier": "^2.8.1",
         "rimraf": "^3.0.2",
-        "ts-jest": "^29.0.3",
         "ts-node": "^10.9.1",
-        "typescript": "^4.9.3"
+        "typescript": "^4.9.5",
```
Comment (Goldziher) [inline package.json:59]: I think you can run `pnpm up --latest` - all should be ok.

File: src/type-factory.ts [hunk 4]
```diff
@@ -17,25 +17,29 @@ import {
...
         | Promise<FactoryDefaults<Partial<T>>>
         | FactoryDefaults<Partial<T>>;
     factory?: FactoryFunction<T>;
     iteration: number;
-};
+}
 
-export class BuildArgProxy {}
+export class BuildArgProxy {
+    // This class is intentionally left empty.
```
Comment (Goldziher) [inline src/type-factory.ts:37]: Perhaps explain this is a sentinel type

File: src/type-factory.ts [hunk 5]
```diff
@@ -17,25 +17,29 @@ import {
...
+}
 
-export class DerivedValueProxy {}
+export class DerivedValueProxy {
+    // This class is intentionally left empty.
```
Comment (Goldziher) [inline src/type-factory.ts:41]: Same

File: vitest.config.ts (hunk 1 above)
Comment (Goldziher) [reply vitest.config.ts:1]: That's fine

**Review by Goldziher — APPROVED** [review]
Looks good. Please ensure both tests and solar work before merging.

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (ghalibansari) [reply .github/workflows/codeql.yml:1]: yes run it manually before commit , thats why this changes are shown

File: package.json (hunk 3 above)
Comment (ghalibansari) [reply package.json:59]: yes, i tried something else it braking, one problem at a time.

File: src/type-factory.ts (hunk 4 above)
Comment (ghalibansari) [reply src/type-factory.ts:37]: sure

**Pull request merged**